### ISSUE TITLE
Query suggestion log and log region mismatch

### ISSUE THREAD

**Issue by greg0ire** [k-yomo/terraform-provider-algolia#63]
### Terraform Version
- Terraform version: v0.14.6
- Provider version: v0.2.6

### Affected Resource(s)
- `algolia_query_suggestions`

### Terraform Configuration Files
Copy-paste your Terraform configurations here
```hcl
resource "algolia_query_suggestions" "main" {
  index_name = format("%s_query_suggestions_%s", var.env_prefix, var.platform)

  source_indices {
    index_name  = local.product_index_name
    min_hits    = 5
    min_letters = 4
  }

  exclude = local.platform_config[var.platform].excluded_query_suggestions

  languages = [local.platform_config[var.platform].language]
}
```

### Expected Behavior

The query suggestion index should get created.

### Actual Behavior
In the past, what happened is that the creation silently failed and then I imported the query suggestion index afterwards.
Now, I'm getting a mysterious message: `Algolia API error [401] The log processing region does not match`

### Steps to Reproduce

I contacted the support and apparently, this has to do with the region we are using for our indexes mismatching the region we use for our logs (and for the whole app). It seems that this plugin targets `query-suggestions.us.algolia.com` when it really should target `query-suggestions.eu.algolia.com` in my case

### Important Factoids

We are a EU-based company.

### References

Here is the latest answer from Algolia support, with a more lengthy explanation:

> Query Suggestions indexes are regionalised, ie, you create an index in a given region (us or eu). This is done targeting a specific host, query-suggestions.us.algolia.com for US and query-suggestions.eu.algolia.com for EU. You can find out more information on this in the documentation [here](https://www.algolia.com/doc/rest-api/query-suggestions/#domain).

 

> We used to not check the regions in the API, enabling you to create Query Suggestions indexes in a region mismatching the log region. We recently added a check on the API to prevent this. That’s why you are receiving [401] The log processing region does not match .

 

> Looking at your dashboard, log region is indeed set to “DE“.

 

> This means that you must be trying to create a new index using query-suggestions.us.algolia.com instead of query-suggestions.eu.algolia.com, (see documentation [here)](https://www.algolia.com/doc/rest-api/query-suggestions/#domain).

 

> When using the dashboard you don’t have to worry about the regions. But since the creation of your index seems to be handled manually (ie, through terraform), you must make sure the domain you use matches the log region.

 

> Additionally, the index “int_query_suggestions_gb“ was created in a different region than the log region. Thus, the index never gets built. Here is an error they found in logs:

> index is being built in different region than where the logs are processed for index "int_query_suggestions_gb" and appID "redacted"

**Comment by greg0ire** [comment]
ping @k-yomo :slightly_smiling_face:

**Comment by k-yomo** [comment]
@greg0ire 
Oh sorry, I missed the notification!
Oh I see, it looks like we need to be able to set the region when we create a query suggestion index.
I'll fix it soon! (hopefully in this week)

**Comment by greg0ire** [comment]
Awesome! Thanks a lot!

**Comment by greg0ire** [comment]
Hi! I have tested your patch, it appears to work on an my non-production apps, modulo some `terragrunt state rm` and `terragrunt import`, but it does not work on my production apps for some reason. I've reached out to Algolia's support team, hopefully they will be able to find the root cause :crossed_fingers: 

Is there a way I could see the HTTP requests performed by the provider? I think that would help a lot with the debugging.
